sphericart/include/dynamic_cuda.hpp

```diff
@@ -0,0 +1,306 @@
+                      << NVRTC::instance().nvrtcGetErrorString(result) << '\n'                     \
+                      << "File: " << __FILE__ << '\n'                                              \
+                      << "Line: " << __LINE__ << '\n';                                             \
+            exit(1);                                                                               \
+        }                                                                                          \
+    } while (0)
+
+#define CUDADRIVER_SAFE_CALL(x)                                                                    \
+    do {                                                                                           \
+        CUresult result = x;                                                                       \
+        if (result != CUDA_SUCCESS) {                                                              \
+            const char* msg;                                                                       \
+            CUDADriver::instance().cuGetErrorName(result, &msg);                                   \
+            std::cerr << "\nerror: " #x " failed with error " << msg << '\n'                       \
+                      << "File: " << __FILE__ << '\n'                                              \
+                      << "Line: " << __LINE__ << '\n';                                             \
+            exit(1);                                                                               \
+        }                                                                                          \
+    } while (0)
+
+#define CUDART_SAFE_CALL(call)                                                                     \
+    do {                                                                                           \
+        cudaError_t cudaStatus = (call);                                                           \
+        if (cudaStatus != cudaSuccess) {                                                           \
+            std::cerr << "CUDA error at " << __FILE__ << ":" << __LINE__ << " - "                  \
+                      << CUDART::instance().cudaGetErrorString(cudaStatus) << '\n'                 \
+                      << "File: " << __FILE__ << '\n'                                              \
+                      << "Line: " << __LINE__ << '\n';                                             \
+            exit(1);                                                                               \
+        }                                                                                          \
+    } while (0)
+
+// Define a template to dynamically load symbols
+template <typename FuncType> FuncType load(void* handle, const char* functionName) {
+    auto func = reinterpret_cast<FuncType>(dlsym(handle, functionName));
```

Comment:
I would add the same `#ifdef` as around `#include <dlfcn.h>`, and then `#error "platform is not supported"` on the other branch